### DISCUSSION TITLE
How to update the current session without signing out

### DISCUSSION THREAD

**Discussion by anzau** [nextauthjs/next-auth#4195] [Help]
Hi there!

I'm currently implementing a profile page where the user changes their image and display name which is attached to the user session and token. Now, I want to update that current session without signing out but I'm unsure how I can go about it? I'm able to update on the database with no problems at all but I'm unable to update the session.

I'm using multiple providers along with a credential provider and I'm verifying the token on certain endpoints so I'm avoiding to use the 'Database' strategy. 

Any help/direction will be appreciated! Thanks!

**Comment by ryanbarr** [comment]
Hey @anzau,

There are likely a number of ways to go about this and ultimately it depends on what patterns you're currently following and which you'd like to follow in the future. I'll recommend two myself:

### A `next-auth` solution
You can utilize the options here: https://next-auth.js.org/getting-started/client#options

Utilizing `refetchInterval` and/or `refetchOnWindowFocus` would allow you to force your sessions to be regularly updated. I am uncertain if this actually refreshes the `user` object in the session, but it is possible.

### A solution utilizing context and state
The solution I would recommend, however, would be to utilize application state. `next-auth` conveniently provides some `user` details upon login to help prevent the need for a second round trip to collect essential details about the user.

However, that doesn't mean that `next-auth` _should_ be responsible for maintaining that user's state after the login completes. I would recommend creating a custom User context. There are a number of resources on using context, but I'd recommend [React's official docs](https://reactjs.org/docs/hooks-reference.html#usecontext) and [Kent C. Dodd's write-up on context](https://kentcdodds.com/blog/how-to-use-react-context-effectively) which includes writing a custom hook.

In the case of Kent's write-up, you would:

1. Create a `<UserProvider>` that you implement in your custom app, then
2. Upon login, you would `dispatch` an event which sets the state of the user in the context, provided by the session object, then
3. In any place you use the user information, you would utilize a custom `useUser()` hook.

This solution then allows you to `dispatch` mutations to the user object to keep the User context updated while not relying on `next-auth` to regularly re-fetch the session.

These solutions are not definitive and there are plenty of other solutions worth exploring. If you're relying strictly on SSR, your desired options may be different as well.
